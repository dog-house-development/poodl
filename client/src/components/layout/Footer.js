import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import GoHome from '../ui/GoHome';

class Footer extends Component {
    render() {
        return (
            <div className="footer">
                <ul>
                    <li>
                        <GoHome className="footer-link" buttonOrLink='link' kind="tertiary" />
                    </li>
                    <li>
                        <a
                            href="https://github.com/dog-house-development/poodl/"
                            className="link tertiary footer-link"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            GitHub
                        </a>
                    </li>
                    <li>
                        <a
                            href="https://github.com/dog-house-development/poodl/wiki/User-Documentation"
                            className="link tertiary footer-link"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            User Documentation
                        </a>
                    </li>
                </ul>
            </div>
        );
    }
}

export default Footer;
