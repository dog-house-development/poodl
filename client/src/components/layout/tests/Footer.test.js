import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import Footer from '../Footer';

configure({ adapter: new Adapter() });

describe('Navbar tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        const props = _.assign({}, _props);
        const state = _.assign({}, _state);
        wrapper = shallow(<Footer {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
