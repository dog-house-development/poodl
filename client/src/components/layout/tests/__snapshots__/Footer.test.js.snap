// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests render should render correctly 1`] = `
<div
  className="footer"
>
  <ul>
    <li>
      <Link
        className="link tertiary footer-link"
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <a
        className="link tertiary footer-link"
        href="https://github.com/dog-house-development/poodl/"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
    <li>
      <a
        className="link tertiary footer-link"
        href="https://github.com/dog-house-development/poodl/wiki/User-Documentation"
        rel="noopener noreferrer"
        target="_blank"
      >
        User Documentation
      </a>
    </li>
  </ul>
</div>
`;
